src/client/dfuse/il/int_posix.c

```diff
@@ -764,10 +780,21 @@ check_ioctl_on_open(int fd, struct fd_entry *entry, int flags)
 	rc = pthread_mutex_lock(&ioil_iog.iog_lock);
 	D_ASSERT(rc == 0);
 
-	if (!ioil_iog.iog_daos_init)
+	if (!ioil_iog.iog_daos_init) {
 		if (!call_daos_init(fd))
 			goto err;
 
+		rc = daos_eq_create(&ioil_eqh);
+		if (rc) {
+			DFUSE_LOG_WARNING("daos_eq_create() failed: "DF_RC"\n", DP_RC(rc));
```

Comment:
As some history here dfuse predates daos so initially the DFUSE_LOG macros were wrappers around the gurt logging and dfuse has always added a \n in the macro automatically.  When daos came along it didn't do this so D_LOG* etc needed the \n.  At some point several years ago however gurt was updated to add the newline in the c function if it was missing so for the D_LOG macros it's optional.

Including rc at the end of the line has always been in the coding standards but we didn't have DF_RC initially and it's use has been sporadic and inconsistent.

The addition of DL_ERROR and friends is an attempt to get on top of the chaos and it handles rc automatically, putting the DF_RC logic in for you, but that relies on the macro adding text to the end of the error message - something it can't do if there's already a new-line there, hence DL_ERROR needs to not have a new-line even though most developers are used to adding one, an obvious cause of annoying issues, hence the check that macros that should not be provided with a new-line do not provide one.

The end result of all this is use DL_ERROR() here, or DHL() if there's a handle available.  There is a handle for the entry here but that's not actually relevant to any failure so if a handle was used here it would be `&ioil_iog`
```suggestion
			DL_ERROR(rc, "daos_eq_create() failed");
```

Reply:
just following what has been done here. im not interested in changing what is here to the new macros atm

Reply:
Then you just need to remove the \n from where you're getting these warnings.

Reply:
ok will do that.

Reply:
there are several other occurences in this file and others where \n is included fwiw..

Reply:
They will be calls to D_ERROR or D_INFO which do not add the newline at the macro level.

Reply:
sigh.. how is one to remember all this

Reply:
This is why I'm trying to ratify it all and add tools to identify mis-use.  See my first comment.